app/assets/javascripts/type_form.js

```diff
@@ -0,0 +1,13 @@
+jQuery(document).ready(function($) {
+  jQuery("[id^=type_attribute_visibility_default]").change(function() {
```

Comment:
Please add a simple class to the check boxes to allow addressing them through that selector, I fear a global id matches selector might be somewhat slower (even if not immediately noticeable).